tools/socktap/autotalks.cpp

```diff
@@ -0,0 +1,229 @@
+#include "autotalks.hpp"
+
+#include <vanetza/access/g5_link_layer.hpp>
+#include <vanetza/common/serialization_buffer.hpp>
+#include <vanetza/access/ethertype.hpp>
+#include <pthread.h>
+#include <iostream>
+
+#include <atlk/sdk.h>
+#include <atlk/v2x.h>
+#include <atlk/v2x_service.h>
+#include <atlk/ddm_service.h>
+#include <atlk/wdm.h>
+#include <atlk/dsm.h>
+#include <atlk/wdm_service.h>
+#include <atlk/log_service.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <extern/ref_sys.h>
+#include <extern/target_type.h>
+#include <extern/time_sync.h>
+
+#ifdef __cplusplus
+}
+#endif
+
+#include "autotalks_link.hpp"
+
+v2x_socket_t* v2x_socket_ptr;
+bool endRxThread = false;
+
+static pthread_t v2x_rx_thread;
+static uint8_t v2x_rx_buffer[2048];
+
+
+int v2x_device_init(void)
+{
+  atlk_rc_t rc;
+  const char* arg[] = {"./test", "enx0002ccf00006"};
+  /** Reference system initialization */
+#ifdef CRATON_2
+  rc = ref_sys_init_ex(1, (char**) arg);
+#else
+  #ifdef SECTON
+    rc = ref_sys_init_ex(2, (char**) arg);
+  #else
+    #error Wrong configuration selected.
+  #endif
+#endif
+
+  // TODO                                                                     //
+  // TODO: Take initialization from Autotalks basic example's main function   //
+  // TODO                                                                     //
```

Comment:
Proposal: Add `#error "Autotalks device initialization code is missing"` so this code does not silently compile and dysfunctional application.